gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionPropsListenerRegistrar.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import com.intellij.openapi.vfs.AsyncFileListener;
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent;
+import java.util.List;
+import org.jetbrains.annotations.Nullable;
+
+public final class VersionPropsListenerRegistrar implements AsyncFileListener {
+
+    private final DebouncingAsyncFileListener debouncedListener;
+
+    VersionPropsListenerRegistrar() {
+        AsyncFileListener versionPropsListener = new VersionPropsFileListener();
+        debouncedListener = new DebouncingAsyncFileListener(versionPropsListener, 250);
```

Comment:
I would probably write this all in one line, also a general style thing is to always use `this.` in constructors for referencing fields (you're normally forced to do this by the constructor params having the same name, but it looks odd if you don't for everything - in constructors only).